components/bl-mapbox-component/src/index.js

```diff
@@ -0,0 +1,224 @@
+import { useEffect, useRef, useState } from 'react';
+
+import mapboxgl from './lib/mapbox';
+import MapboxDirections from './lib/mapbox-directions';
+import MapboxGeocoder from './lib/mapbox-geocoder';
+import { createActions } from './actions';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const defaultValues = {
+  START_POS           : { lat: 0, lng: 0 },
+  MAP_STYLE           : 'mapbox://styles/mapbox/streets-v11',
+  ZOOM                : 10,
+  PROJECTION          : 'mercator',
+  LOWER_ATMOSPHERE    : '#BAD2EB',
+  UPPER_ATMOSPHERE    : '#245CDF',
+  ATMOSPHERE_THICKNESS: 0.2,
+  SPACE_COLOR         : '#0B0B19',
+  STAR_INTENSITY      : 0.2,
+};
+
+export default function Mapbox({ component, eventHandlers }) {
+  const mapContainerRef = useRef(null);
+  const mapRef = useRef(null);
+
+  const [polygonsObject, setPolygonsObject] = useState([]);
+
+  const { onClick, onMarkerClick, onPolygonClick, onDeterminingGeoposition, onPan } = eventHandlers;
+
+  const {
+    markers, polygons, center, zoom, fullScreen, accessToken, projection, lowerAtmosphere, upperAtmosphere,
+    atmosphereThickness, spaceColor, starIntensity, mapStyle, navigation, searchBar, geolocation, directions,
+    display, classList,
+  } = component;
+
+  if (!display) {
+    return null;
+  }
```

Comment:
That caused an errors

Reply:
This is definitely a solution, but not which I expected to see :)

Any return from component should be **after** all hooks - that's all. No need in additional component <Map/>

If you want to go deeper - just google the error.

Reply:
We need an additional component to avoid errors from mapbox, when `display` property is `false`. 
`Invalid type: 'container' must be a String or HTMLElement.`

Also when we toggle `display` property from `true` to `false` and then from `false` to `true` map isn't loading. (This issue exists in Leaflet map).